src/views/blocks/ImageBlock.php

```diff
@@ -8,19 +8,38 @@
  * @param $this->varValue('align');
  * @param $this->varValue('fluid');
  * @param $this->varValue('image');
+ * @param $this->cfgValue('lazyload');
  *
  * @var $this \luya\cms\base\PhpBlockView
  */
+use luya\lazyload\LazyLoad;
 
 $align = $this->varValue('align', 'left');
 ?>
 
-<?php if ($this->extraValue('image', false)): ?>
+<?php if ($image = $this->extraValue('image', false)): ?>
     <div class="image">
         <figure class="figure d-block <?php if ($align === 'left'): ?>text-left<?php elseif ($align === 'right'): ?>text-right<?php else: ?>text-center<?php endif; ?>">
-            <img src="<?= $this->extraValue('image')->source ?>" class="figure-img img-fluid" <?php if ($this->extraValue('image')->caption): ?>alt="<?= $this->extraValue('image')->caption ?>"<?php endif; ?>>
-            <?php if ($this->varValue('showCaption', false) && $this->extraValue('image')->caption): ?>
-                <figcaption class="figure-caption"><?= $this->extraValue('image')->caption ?></figcaption>
+            <?php
+            $caption = !empty($image->caption) ? $image->caption : '';
+            if ($this->cfgValue('lazyload', false)):
+                $options['src'] = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==';
+                if (!empty($caption)) {
+                    $options['alt'] = $caption;
+                    $options['title'] = $caption;
+                }
+                echo LazyLoad::widget([
+                    'src' => $image->source,
+                    'extraClass' => 'figure-img img-fluid',
+                    'width' => $image->itemArray['resolution_width'],
+                    'height' => $image->itemArray['resolution_height'],
+                    'options' => $options
+                ]);
+            else: ?>
+                <img src="<?= $image->source ?>" class="figure-img img-fluid"<?php if (!empty($caption)): ?> alt="<?= $caption ?>"<?php endif; ?>>
```

Comment:
!empty($caption) should be fine instead of (!empty($caption))

Reply:
phpstorm tells me there's an error... [see here](https://i.imgur.com/CmqZ8uZ.png)

Reply:
@JohnnyMcWeed i am sorry, indeed this wont work in template syntax